boards/xtensa/intel_s1000_crb/support/create_board_img.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This script will add specific headers and FW load message to tell
+# intel_s1000 ROM bootloader to load the firmware
+# Usage python3 ${ZEPHYR_BASE}/scripts/create_board_img.py
+#                    -i in_file.bin -o out_file.bin
+#                    [-m|--no_sram] [-c|--no_l1cache] [-t|--no_tlb]
+#                    [-x|--no_exec] [-s|--no_sha] [-k|--clk_sel]
+
+import argparse
+import os
+import sys
+import struct
+
+help_text="""
+The flash memory needs to have specific headers and fw load message to tell
+the ROM bootloader to load firmware.
+
+To run script use command:
+
+create_spi_flash.py -i in_file.bin -o out_file.bin
+                    [-m|--no_sram] [-c|--no_l1cache] [-t|--no_tlb]
+                    [-x|--no_exec] [-s|--no_sha] [-k|--clk_sel]
+"""
+
+FLASH_PART_TABLE_OFFSET	= (0x1000)
+FLASH_SECTOR_SIZE	= (4 * 1024)
+FLASH_MAGIC_WORD	= (0x30504353)
+
+SRAM_SIZE		= (1024*1024*4)
+MAX_FLASH_FILE_SIZE	= (1024*1024*2)
+SIZEOF_IPC64		= (64)
+ROM_CONTROL_MEMWRITE	= (0x11)
+ROM_CONTROL_LOADFW	= (0x2)
+ROM_CONTROL_EXEC	= (0x13)
+
+FW_LOAD_NO_L1_CACHE	= (1 << 29)
+FW_LOAD_NO_SRAM_FLAG	= (1 << 28)
+FW_LOAD_NO_TLB_FLAG	= (1 << 27)
+FW_LOAD_NO_EXEC_FLAG	= (1 << 26)
+FW_LOAD_NO_SHA_FLAG	= (1 << 25)
+FW_LOAD_CLK_SEL		= (1 << 21)
+
+FW_LOAD_ADDR		= (0xbe000000)
```

Comment:
Can the load address be made configurable (through a command-line option, for example)? As we discussed, the bootloader's footprint is reduced to under 64KB, we would want to load the bootloader image on to LPSRAM.

Reply:
sure. will see.

Reply:
Done.